### DISCUSSION TITLE
How oh-my-posh get NodeJS version

### DISCUSSION THREAD

**Discussion by damadorPL** [JanDeDobbeleer/oh-my-posh#5857] [Troubleshoot] [answered]
node -v
v23.1.0

return the right version that is installed globally 

started to experiment with nvm for windows and installed version 20.18.0 then after few dyas uninstalled it - removed all folders created even reinstalled node again 

now everytime I start powershell in node folder it show  20.18.0 ...

how it check for version of node ? 

block that I use 

                {
		  "type": "node",
		  "style": "diamond",
		  "powerline_symbol": "",
		  "foreground": "#FFE873",
		  "background": "",
		  "template": "  {{ .Full }} "
		}

no matter that i put in "template" always return v20 (partial or full) 

there is no env folders in windows from nvm - only C:\Program Files\nodejs\ is in system variables 
		
Win 11 

nodes for python (4 global , multiple from condas and venv) always works ad return proper version but somehow cant figure how Node get version info

**Comment by lewis-yeung** [comment]
@damadorPL It's due to the (global) caching mechanism of a language segment. Please wait for #5859 which will allow disabling the cache.

**Reply by JanDeDobbeleer** [reply]
@damadorPL it's merged so you'll be able to set `"cache_duration": "none"` in the `node` segment's `properties` right away.

**Reply by damadorPL** [reply]
```
                {
		  "type": "node",
		  "style": "diamond",
		  "powerline_symbol": "",
		  "foreground": "#FFE873",
		  "background": "",
		  "template": "  {{ .Full }} ",
		  "properties": {
			"cache_duration": "none"
		  }
		}
```

![Screenshot 2024-11-08 185548](https://github.com/user-attachments/assets/2be817c7-2d73-44c9-83ff-d57aae0722a2)

added and now got

**Reply by damadorPL** [reply]
https://gist.github.com/damadorPL/1367c16f6b096c62646f4cc2d8c52b9e  and the whole theme

**Reply by JanDeDobbeleer** [reply, marked as answer]
@damadorPL you need to clear the cache as well for it to get picked up: `oh-my-posh cache clear`

**Reply by damadorPL** [reply]
After 2nd terminal restart / and OS itself  first ( context ...) error is gone, but node version still incorrect

**Reply by damadorPL** [reply]
![Screenshot 2024-11-08 192319](https://github.com/user-attachments/assets/c68fc085-c0b7-4bbe-94ac-a55c37644a8c)

fresh directory , fresh gh clone ... 

the 20.18 version was installed previously with node version maanger for windowns - its completly uninstalled and somehow that version is still detected by omp

**Reply by JanDeDobbeleer** [reply]
@damadorPL you selected an answer, are you still experiencing the issue? If so, can you share your configuration?

**Reply by damadorPL** [reply]
oh-my-posh cache clear  fixed it and I did it after previous followups - sorry for bothering
